Azure/cloud-init.txt

```diff
@@ -0,0 +1,167 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Description: This script will install all dependencies
+# required by clear containers and metrics/tests.
+#
+# OS: ubuntu 16.04
+
+# Add repository key
+wget -nv https://download.opensuse.org/repositories/home:clearcontainers:clear-containers-3/xUbuntu_16.04/Release.key -O Release.key
+apt-key add - < Release.key
+apt-get update
+
+# Add repositories and install runtime
+echo "Installing Clear Containers"
+sh -c "echo 'deb http://download.opensuse.org/repositories/home:/clearcontainers:/clear-containers-3/xUbuntu_16.04/ /' > /etc/apt/sources.list.d/cc-runtime.list"
+apt-get update
+apt-get install -y cc-runtime
+
+
+# Install docker
+echo "Instaling docker"
+apt-get install -y \
+        apt-transport-https \
+        ca-certificates \
+        curl \
+        software-properties-common
+
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+apt-key fingerprint 0EBFCD88
+
+add-apt-repository \
+        "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+        $(lsb_release -cs) \
+        stable"
+
+apt-get update
+apt-get install -y docker-ce
+
+# Install tests dependencies
+apt-get install -y \
+	bc \
+	smem \
+	apache2-utils \
+	iperf
+
+# Install thin tools for devicemapper congifuration
+apt-get install -y thin-provisioning-tools
+
+# Configre the clear containers runtime using devicemapper as
+# storage driver
+mkdir -p /etc/systemd/system/docker.service.d/
+cat << EOF | sudo tee /etc/systemd/system/docker.service.d/clr-containers.conf
+[Service]
+ExecStart=
+ExecStart=/usr/bin/dockerd -D --add-runtime cc-runtime=/usr/bin/cc-runtime --default-runtime=cc-runtime
+EOF
+
+# Add devicemapper configuration
+mkdir -p /etc/docker
+
+mkdir -p /etc/lvm/profile
+cat << EOF | sudo tee /etc/docker/daemon.json
+{
+"storage-driver": "devicemapper",
+"storage-opts": [
+        "dm.directlvm_device=/dev/sdc",
+        "dm.thinp_percent=95",
+        "dm.thinp_metapercent=1",
+        "dm.thinp_autoextend_threshold=80",
+        "dm.thinp_autoextend_percent=20",
+        "dm.directlvm_device_force=true"
+]
+}
+EOF
+
+# Restart docker with devicemapper changes
+systemctl daemon-reload
+#systemctl restart docker
+
+# Clone tests/metrics repository
+git clone https://github.com/clearcontainers/tests.git
+
+# Update kernel
+echo deb http://archive.ubuntu.com/ubuntu/ xenial-proposed restricted main multiverse universe >> /etc/apt/sources.list
+apt-get update
+apt-get install -y linux-azure
+
+# Install Kubernetes and kubeadm
+apt-get install -y ebtables ethtool
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+cat << EOF > /etc/apt/sources.list.d/kubernetes.list
+deb http://apt.kubernetes.io/ kubernetes-xenial main
+EOF
+apt-get update
+apt-get install -y kubelet kubeadm kubectl
```

Comment:
Should we not pick a specific version

@chavafg uses https://github.com/clearcontainers/tests/blob/master/integration/kubernetes/setup.sh#L27
```
sudo -E apt install -y docker.io kubelet=1.6.7-00 kubeadm=1.6.7-00 kubectl=1.6.7-00
```

Reply:
@mcastelino: Very good point, we should use 1.7.5 with CRI-O 1.0

Reply:
@1.7.5 is compatible with CC, there is a PR opened for this #526